examples/gno.land/r/demo/profile/profile_test.gno

```diff
@@ -0,0 +1,157 @@
+package profile
+
+import (
+	"std"
+	"testing"
+
+	"gno.land/p/demo/testutils"
+)
+
+const (
+	DisplayName        = "DisplayName"
+	Homepage           = "Homepage"
+	Bio                = "Bio"
+	Age                = "Age"
+	AvailableForHiring = "AvailableForHiring"
+	InvalidField       = "InvalidField"
+)
+
+func TestStringFields(t *testing.T) {
+	addr := testutils.TestAddress("alice")
+	std.TestSetRealm(std.NewUserRealm(addr))
+
+	// Get before setting
+	name := GetStringField(addr, DisplayName, "anon")
+	assertSameString(t, name, "anon")
+
+	// Set
+	if err := SetStringField(DisplayName, "Alice foo"); err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	if err := SetStringField(Homepage, "https://example.com"); err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	// Get after setting
+	name = GetStringField(addr, DisplayName, "anon")
+	homepage := GetStringField(addr, Homepage, "")
+	bio := GetStringField(addr, Bio, "42")
+
+	assertSameString(t, name, "Alice foo")
+	assertSameString(t, homepage, "https://example.com")
+	assertSameString(t, bio, "42")
+}
+
+func TestIntFields(t *testing.T) {
+	addr := testutils.TestAddress("bob")
+	std.TestSetRealm(std.NewUserRealm(addr))
+
+	// Get before setting
+	age := GetIntField(addr, Age, 25)
+	assertSameInt(t, age, 25)
+
+	// Set
+	if err := SetIntField(Age, 30); err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	// Get after setting
+	age = GetIntField(addr, Age, 25)
+	assertSameInt(t, age, 30)
+}
+
+func TestBoolFields(t *testing.T) {
+	addr := testutils.TestAddress("charlie")
+	std.TestSetRealm(std.NewUserRealm(addr))
+
+	// Get before setting
+	hiring := GetBoolField(addr, AvailableForHiring, false)
+	assertSameBool(t, hiring, false)
+
+	// Set
+	if err := SetBoolField(AvailableForHiring, true); err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	// Get after setting
+	hiring = GetBoolField(addr, AvailableForHiring, false)
+	assertSameBool(t, hiring, true)
+}
+
+func TestInvalidStringField(t *testing.T) {
+	addr := testutils.TestAddress("dave")
+	std.TestSetRealm(std.NewUserRealm(addr))
+
+	err := SetStringField(InvalidField, "test")
+	if err == nil {
+		t.Error("expected error for invalid field, got nil")
+	}
+}
+
+func TestInvalidIntField(t *testing.T) {
+	addr := testutils.TestAddress("eve")
+	std.TestSetRealm(std.NewUserRealm(addr))
+
+	err := SetIntField(InvalidField, 123)
+	if err == nil {
+		t.Error("expected error for invalid field, got nil")
+	}
+}
+
+func TestInvalidBoolField(t *testing.T) {
+	addr := testutils.TestAddress("frank")
+	std.TestSetRealm(std.NewUserRealm(addr))
+
+	err := SetBoolField(InvalidField, true)
+	if err == nil {
+		t.Error("expected error for invalid field, got nil")
+	}
+}
+
+func TestMultipleProfiles(t *testing.T) {
+	addr1 := testutils.TestAddress("user1")
+	addr2 := testutils.TestAddress("user2")
```

Comment:
you can register some users globally at the package-level since you're creating addresses on each tests; it will improve readability.